### ISSUE TITLE
"Zero-day streak" gives 21-day streak achievement

### ISSUE THREAD

**Issue by TwasBrillig1** [HabitRPG/habitica#2578]
Hi, today I tried to restore a daily's streak to 0 (I checked it off yesterday, but today I realized I hadn't actually done the task). When I entered 0 in the "restore streak" box and clicked Save, the streak counter displayed as "-1". I checked it off for today, and a dialogue box appeared saying that I stacked my 21-day streak achievement. The daily's streak counter is now blank (zero). 

(Plus, yesterday's accidental check was day 21 of a streak, so I now have two more of those achievements than I deserve, oops!) 

Browser: Chrome Version 32.0.1700.76 m
User ID: 3ab30744-c44f-491a-b69e-c9eeefa5f24e
I didn't see any errors in the console. 

Please let me know if you need any more info - Thanks!
##

**Comment by deilann** [comment]
Ahah, I think I know what that is. I'm guessing the system looks for multiples of 21.

0 is, in fact, a multiple of 21. I'll look into it.

Thanks!

**Comment by Alys** [comment]
I've posted this to Aspiring Coders.

If someone decides to work on this, please comment here to claim it. Ask for help here if you need it!

**Comment by librarianmage** [comment]
Has this been taken care of?

**Comment by Alys** [comment]
I tested by repeating the steps from the OP and it still behaves like that. However, we probably want to put this on hold for now. We'd need to modify code under `common/script` for this, which is best done after version 3 of the API has been released (in development now).

**Comment by librarianmage** [comment]
@Alys Yep!

**Comment by Alys** [comment]
I'm most of the way through refactoring common/script/ops/scoreTask.js which contains the code for this, so I'm assigning this to myself.

**Comment by khipkin** [comment]
@Alys are you still planning to work on this?

**Comment by Alys** [comment]
@khipkin I was intending to do it, but was waiting on https://github.com/HabitRPG/habitrpg/pull/7607 which I haven't completed yet because I had some questions about what tests were appropriate, and I've been forgetting to follow up on that. However I've got holidays coming up so I'll probably just write the tests I have in mind. Once the refactored scoreTask.js is live, other changes to it will be easier, including this one. If you want to work on this, I'll happily relinquish it to you. I clearly am not good at getting through issues promptly. :-/

**Comment by lemoness** [comment]
Is this still an issue after the redesign?

**Comment by Alys** [comment]
Yes, same issue and cause.

**Comment by Alys** [comment]
I'm working on this.
